src/main/java/org/broadinstitute/hellbender/utils/collections/NestedIntegerArray.java

```diff
@@ -82,6 +82,64 @@ public T get(final int... keys) {
         return (T)myData[keys[numNestedDimensions]];
     }
 
+    @SuppressWarnings("unchecked")
+    private T leaf(final int key, final Object[] array){
+        return array == null ? null : (T) array[key];
```

Comment:
It's not true that (T) is just a cast to `Serializable` -- it's a cast to whatever the actual type of `T` happens to be. You can see this by running the following code:

    public class Foo2<T extends Object> {
        private Object o;

        public Foo2(Object o ) {
            this.o = o;
        }

        public T get() {
            return (T)o;
        }

        public static void main ( String[] args ) {
            Foo2<String> f = new Foo2<>(new Integer(5));
            System.out.println(f.get());
        }
    }

Suppressing warnings and allowing `ClassCastException`s to leak is nastier than detecting them and throwing a meaningful error message.